### ISSUE TITLE
PS 1.7.6.4 setAllowAnonymous method not found on AdminSearchController

### ISSUE THREAD

**Issue by DragoshDX** [PrestaShop/PrestaShop#19406]
#### Describe the bug

Hello. 
While trying to rebuild the search index manually, I get: 

Attempted to call an undefined method named "setAllowAnonymous" of class "AdminSearchController".

Doesn't matter if I choose to rebuild the whole index or just add the missing products (obviously the same issue is with the cron URL)

Go to Shop Parameters > Search and hit any rebuild index link. It should give you the error. 

Based on what I found in the code the method was removed but the call to it in the rebuild index feature has not. 

AdminSearchController.php: 

```public function init()
    {
        if ($this->isCronTask()
            && substr(
                _COOKIE_KEY_,
                static::TOKEN_CHECK_START_POS,
                static::TOKEN_CHECK_LENGTH
            ) === Tools::getValue('token')
        ) {
            $this->setAllowAnonymous(true);
        }

        parent::init();
    }
```

After I remove the call, everything just works and my products are in fact added to the index and I can find them via site search. 

Not sure if its an issue on 1.7.7 anymore.

* PrestaShop version:  1.7.6.4
* PHP version: 7.0

**Comment by matks** [comment]
Strange, function `isAnonymousAllowed ` has been added in 1.7.6.2 https://github.com/PrestaShop/PrestaShop/pull/15855

**Comment by matks** [comment]
@DragoshDX I can see the function in the codebase for tag 1.7.6.4 : https://github.com/PrestaShop/PrestaShop/blob/1.7.6.4/classes/controller/AdminController.php#L4860

Can you check again on your shop ? Maybe someone modified the core code of prestashop so it's not up-to-date with github ?

**Comment by DragoshDX** [comment]
Looks like you are right, but my AdminController file simply doesn't have that method... 
It may be before we added our code to git and I can't see any changes to see if someone might have intentionally deleted the method.

Maybe it will get fixed when we upgrade to 1.7.7 which I plan for 2020. 

Anyway, if this is just me and the method is there, I guess we can close this?

**Comment by matks** [comment]
> Looks like you are right, but my AdminController file simply doesn't have that method...
> It may be before we added our code to git and I can't see any changes to see if someone might have intentionally deleted the method.

You can easily add the method again into the code 😄 it's a small one. It's intented to allow unauthenticated users to use BO controllers (for example cron jobs).

> Maybe it will get fixed when we upgrade to 1.7.7 which I plan for 2020.

If your file has been changed, the Upgrade Assistant will detect it and flag it as a blocking item for the upgrade. It compares file checksums to find this.

> Anyway, if this is just me and the method is there, I guess we can close this?

Yes indeed.

If you need further help, it might be easier on the Slack channel 😉 https://github.com/PrestaShop/open-source/tree/master/slack
